test/unit/sample_type_test.rb

```diff
@@ -911,6 +911,61 @@ def setup
     assert_equal updated_at, sample.updated_at
   end
 
+  test 'adding a creator' do
+    User.current_user = @person.user
+    creator = Factory :person
+    sample_type = Factory(:simple_sample_type, contributor: @person)
+    params = { creator_ids: [creator.id] }
+    assert_difference('sample_type.creators.count') do
+      assert_difference('AssetsCreator.count') do
+        sample_type.update!(params)
+      end
+    end
+    sample_type.save!
+  end
+
+  test 'updating a creator' do
+    User.current_user = @person.user
+    # Set creator
+    creator = Factory :person
+    sample_type = Factory(:simple_sample_type, contributor: @person)
+    params = { creator_ids: [creator.id] }
+    sample_type.update!(params)
+    # Update creator
+    new_creator = Factory :person
+    params = { creator_ids: [new_creator.id] }
+    assert_no_difference('AssetsCreator.count') do
+      assert_no_difference('sample_type.creators.count') do
+        sample_type.update!(params)
+      end
+    end
+    assert_not_equal sample_type.creators.first, creator
+    assert_equal sample_type.creators.first, new_creator
```

Comment:
check the full array, with just `assert_equal [new_creator], sample_type.creators`

then you only need one check, and will catch an unexpected creator being added